Adds email address to SlackUser object

pallan opened the pull request:
Adds the user's email address to the available attributes on a SlackUser. Slack always requires an email address, as far as I can tell, for user accounts so this should be safe to be here.  This could be a potentially breaking change since it changes the initializer for SlackUser though.

This is related to #22. 

@kenjij 
cc/ @byroot @Sirupsen

byroot commented:
:heart:

jimmycuadra commented:
Somewhat similar to #47, as well. I don't really like the idea of having non-essential metadata as a first-class attribute of SlackUser, but maybe email is useful enough to make this change worthwhile. To avoid making it a breaking change, let's just have the constructor pull email out of `raw_data` itself.

pallan commented:
Sounds like a plan.  Will update

pallan commented:
Updated the email to be pulled from `raw_data`.

pallan commented:
I also updated `raw_data` in SlackUser to `metadata` so it matches the interface of `Lita::User`. I added in a accessor method for `raw_data` to maintain backward compatibility.

jimmycuadra commented:
Looking good. Let's use `alias_method` right below the new attribute declaration, instead of defining `raw_data` manually, so it's more clear that they're the same thing.

pallan commented:
Sure, somedays I want to use `alias_method` and somedays I don't :-)  Updated to `alias_method`.

jimmycuadra commented:
Released: https://github.com/kenjij/lita-slack/releases/tag/v1.7.0

pallan commented:
Thank you very much @jimmycuadra!
